crates/rune/src/diagnostics/emit.rs

```diff
@@ -122,8 +122,8 @@ impl Diagnostics {
         &self,
         out: &mut O,
         sources: &Sources,
-        debug_info: Option<&DebugInfo>,
-        context: Option<&Context>,
+        unit: &Unit,
+        context: &Context,
```

Comment:
I've opted to just pass in the `Unit` and `Context` directly. If these are not available, you'd use `emit`. But ultimately this should probably be made into a builder which allows you to more freely associate build elements that are available.